static/app/views/settings/account/notifications/notificationSettingsByTypeV2.tsx

```diff
@@ -0,0 +1,338 @@
+    if (!matchedOption) {
+      if (defaultSettings) {
+        defaultValue = defaultSettings.typeDefaults[notificationType];
+        // return {[notificationType]: defaultSettings.typeDefaults[notificationType]};
+      }
+    } else {
+      defaultValue = matchedOption.value;
+    }
+    // if we have child types, map the default
+    const childTypes: string[] = typeMappedChildren[notificationType] || [];
+    const childTypesDefaults = Object.fromEntries(
+      childTypes.map(childType => [childType, defaultValue])
+    );
+
+    return {
+      [notificationType]: defaultValue,
+      ...childTypesDefaults,
+    };
+  }
+
+  getProviderInitialData(): {[key: string]: string[]} {
+    const {notificationProviders, defaultSettings} = this.state;
+    // user has no settings saved so use default
+    if (notificationProviders.length === 0 && defaultSettings) {
+      return {provider: defaultSettings.providerDefaults};
+    }
+    const providers = notificationProviders
+      .filter(option => option.scopeType === 'user' && option.value === 'always')
+      .map(option => option.provider);
+    return {provider: providers};
+  }
+
+  getFields(): Field[] {
+    const {notificationType} = this.props;
+
+    const help = isGroupedByProject(notificationType)
+      ? t('This is the default for all projects.')
+      : t('This is the default for all organizations.');
+
+    const fields: Field[] = [];
+    // if a quota notification is not disabled, add in our dependent fields
+    // but do not show the top level controller
+    if (notificationType === 'quota') {
+      fields.push(
+        ...QUOTA_FIELDS.map(field => ({
+          ...field,
+          type: 'select' as const,
+          getData: data => {
+            return {
+              type: field.name,
+              scopeType: 'user',
+              scopeIdentifier: ConfigStore.get('user').id,
+              value: data[field.name],
+            };
+          },
+        }))
+      );
+    } else {
+      const defaultField: Field = Object.assign(
+        {},
+        NOTIFICATION_SETTING_FIELDS[notificationType],
+        {
+          help,
+          defaultValue: 'always',
+          getData: data => {
+            return {
+              type: notificationType,
+              scopeType: 'user',
+              scopeIdentifier: ConfigStore.get('user').id,
+              value: data[notificationType],
+            };
+          },
+        }
+      );
+      fields.push(defaultField);
+    }
+
+    return fields;
+  }
+
+  getProviderFields(): Field[] {
+    const {notificationType} = this.props;
+    const {organizationIntegrations} = this.state;
+    const defaultField = Object.assign({}, NOTIFICATION_SETTING_FIELDS.provider, {
+      choices: organizationIntegrations
+        .map(organizationIntegration => [
+          organizationIntegration.provider.slug,
+          organizationIntegration.provider.name,
+        ])
+        .concat([['email', 'Email']]),
```

Comment:
nit - chaining a concat onto a map is a bit funky
```suggestion
      choices: [
      	...organizationIntegrations
	        .map(organizationIntegration => [
	          organizationIntegration.provider.slug,
	          organizationIntegration.provider.name,
	        ]),
	        ['email', 'Email'],
        ]
```